app/labor/markdown_parser.rb

```diff
@@ -18,7 +19,7 @@ def finalize(link_attributes: {})
     sanitized_content = sanitize_rendered_markdown(html)
     begin
       parsed_liquid = Liquid::Template.parse(sanitized_content)
-      html = markdown.render(parsed_liquid.render)
+      html = markdown.render(parsed_liquid.render(source: @source))
```

Comment:
In the liquid world, this is known as an [Assignment](https://github.com/Shopify/liquid/wiki/Liquid-for-Programmers#using-assignments-in-templates). It allows us to access other data inside the tag. It is accessible in `context.environments` which I made a helper to quickly access. I'm thinking of this as the "environment the liquid tag is being rendered in" or "the source (`Article`, `Comment`, etc.) where the liquid tag is being rendered".

This enables us to do context (`Article`, `Comment`, etc.) specific rendering of liquid tags. The same liquid tag could render differently in different contexts/environments.

No other liquid tag uses this yet because this the first "context-specific liquid tag". Very likely it won't be the last 🙃 . I may use this same concept here to validate liquid tags and which contexts they're permitted in. TBD in the next PR when I add role restrictions and validations to liquid tags.

If you're like me and happen to peruse this `MarkdownParser` or other files around rending liquid tags, you'll find other areas where we could improve this entire process. I'm open to other avenues! I chose this as the least invasive and most impactful 🙈 .